hwmv2: port NXP ls1046a SoC and rdb board to v2

JiafeiPan opened the pull request.

JiafeiPan committed:
hwmv2: port nxp_ls1046ardb board to V2

Port NXP ls1046ardb board to hardware model v2.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
hwmv2: port NXP SoC LS1046A to V2

Port NXP SoC LS1046A to hardware model v2.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

nordicjm commented on boards/nxp/ls1046ardb/board.yml:
```diff
@@ -0,0 +1,8 @@
+  vendor: nxp
+  socs:
+  - name: ls1046a
+    variants:
+    - name: smp_2cores
+    - name: smp_4cores
```
non-blocking but in other boards smp is a variant then 2cores and 4cores are variants of that

nordicjm commented on boards/nxp/ls1046ardb/doc/index.rst:
```diff
@@ -67,15 +67,15 @@ Other hardware features have not been enabled yet for this board.
 The default configuration can be found in the defconfig file for NON-SMP:
 
-        ``boards/arm64/nxp_ls1046ardb/nxp_ls1046ardb_defconfig``
+    :zephyr_file:`boards/arm64/nxp_ls1046ardb/nxp_ls1046ardb_defconfig`
```
links are all wrong

JiafeiPan replied:
fixed, thanks.

nordicjm commented on boards/nxp/ls1046ardb/doc/index.rst:
```diff
@@ -91,7 +91,7 @@ kernel tests on LS1046A RDB board. For example, with the :zephyr:code-sample:`sy
    :zephyr-app: samples/synchronization
    :host-os: unix
-   :board: nxp_ls1046ardb
+   :board: nxp_ls1046ardb/ls1046a
```
the SoC can be excluded from these build commands since there is only one

JiafeiPan replied:
Fixed, thanks.

nordicjm commented on soc/nxp/layerscape/Kconfig:
```diff
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+
+rsource "*/Kconfig"
```
missing guard

JiafeiPan replied:
Fixed, thanks.

nordicjm commented on soc/nxp/layerscape/Kconfig.soc:
```diff
@@ -0,0 +1,13 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config SOC_FAMILY_LAYERSCAPE
+	bool
+
+config SOC_FAMILY
+	default "nxp_layerscape" if SOC_FAMILY_LAYERSCAPE
```
either:
```suggestion
config SOC_FAMILY_NXP_LAYERSCAPE
	bool

config SOC_FAMILY
	default "nxp_layerscape" if SOC_FAMILY_NXP_LAYERSCAPE
```
or
```suggestion
config SOC_FAMILY_LAYERSCAPE
	bool

config SOC_FAMILY
	default "layerscape" if SOC_FAMILY_LAYERSCAPE
```

JiafeiPan replied:
Changed to use SOC_FAMILY_NXP_LAYERSCAPE, thanks.

nordicjm commented on soc/nxp/layerscape/ls1046a/Kconfig.soc:
```diff
@@ -0,0 +1,31 @@
+	bool
+	select SOC_SERIES_LS1046A
+	help
+		NXP Layerscape ls1046a
```
help text indentation is 1 tab followed by 2 spaces

JiafeiPan replied:
Fixed, thanks.

nordicjm commented on soc/nxp/layerscape/ls1046a/Kconfig.soc:
```diff
@@ -0,0 +1,31 @@
+	help
+		NXP Layerscape ls1046a
+
+config SOC
+	default "ls1046a"
+
+config SOC_SERIES
+	default "ls1046a"
```
missing if statements

JiafeiPan replied:
fixed, thanks.

nordicjm commented on soc/nxp/layerscape/ls1046a/Kconfig.soc:
```diff
@@ -0,0 +1,31 @@
+	bool
+
+config SOC_PART_NUMBER
+	default "LS1046A" if  SOC_PART_NUMBER_LS1046A
+	default "LS1026A" if  SOC_PART_NUMBER_LS1026A
```
```suggestion
	default "LS1046A" if SOC_PART_NUMBER_LS1046A
	default "LS1026A" if SOC_PART_NUMBER_LS1026A
```

JiafeiPan replied:
fixed, thanks.

JiafeiPan commented:
Addressed all comments, thanks.

nordicjm commented on boards/nxp/ls1046ardb/board.yml:
```diff
@@ -0,0 +1,8 @@
+  vendor: nxp
+  socs:
+  - name: ls1046a
+    variants:
+    - name: smp_2cores
+    - name: smp_4cores
```
Going by what we have in other boards and https://github.com/zephyrproject-rtos/zephyr/pull/69400/commits/b3720c7bf78bb5dc72e6a86a6b0570040988bd8a can you change this so smp is a variant which enables either 2 or 4 cores then have the other as a variant of that, so the boards would be `nxp_ls1046ardb/ls1046a`, `nxp_ls1046ardb/ls1046a/smp`, and (`nxp_ls1046ardb/ls1046a/smp/2cores` or `nxp_ls1046ardb/ls1046a/smp/4cores`)

JiafeiPan replied:
Changed, please review again, thanks.

JiafeiPan replied:
@nordicjm, I updated but I found currently there is document building failure, but I don't know why we have the failure, it seems it is not caused by this PR, thanks.

nordicjm replied:
It is caused by this PR, you need to remove the old arm64 boards_legacy documentation index file if the folder is empty and remove that file being listed by the index.rst file

JiafeiPan replied:
Rebased and added a new commit to clean up empty directory: https://github.com/zephyrproject-rtos/zephyr/pull/69416/commits/cb4fecf9bb826bf01db32f0568dc30c46b4e0d04

nordicjm commented on boards/nxp/ls1046ardb/board.yml:
```diff
@@ -0,0 +1,10 @@
+  socs:
+  - name: ls1046a
+    variants:
+    - name: smp
+      variants:
+      - name: 2cores
+      - name: 4cores
```
no, smp would be one of these e.g. building for `nxp_ls1046ardb/ls1046a/smp` would give you the 2 core version and building for `nxp_ls1046ardb/ls1046a/smp/4cores` would give you the 4 core version, so the `_smp_2cores` filenames would change to `_smp`, and only differences for the 4core ones would remain in the `_smp_4cores` files, which assumingly (I have not looked at the files) would only be the defconfig

nordicjm replied:
And the way the hwmv2 configuration works is the base configuration will be applied first, that is `nxp_ls1046ardb_defconfig`, then `nxp_ls1046ardb_ls1046a_defconfig` will be applied (does not exist so can be ignored) then `nxp_ls1046ardb_ls1046a_smp_defconfig` will be applied, then finally `nxp_ls1046ardb_ls1046a_smp_4cores_defconfig`, so anything in the base files does not need to be set again in the files overlayed on top (note: defconfig files only work like this, only one .dts file is used)

nordicjm replied:
See https://github.com/zephyrproject-rtos/zephyr/tree/collab-hwm/boards/synopsys/nsim for an example

JiafeiPan replied:
@nordicjm , thanks for your detail explanation, have updated, please review again.

JiafeiPan committed:
hwmv2: clean up arm64 soc and board empty directory

There is no legacy ARM64 SoC and Board available, so delete
the arm64 directory.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

nordicjm commented on boards/nxp/ls1046ardb/ls1046ardb_ls1046a_defconfig:
```diff
@@ -1,9 +1,7 @@
+# Copyright 2024 NXP
```
in ls1046ardb_ls1046a.yaml remove nxp_ from identifier

JiafeiPan replied:
fixed, thanks.

nordicjm commented on boards/nxp/ls1046ardb/ls1046ardb_ls1046a_smp.yaml:
```diff
@@ -1,4 +1,4 @@
-identifier: nxp_ls1046ardb_smp_2cores
+identifier: nxp_ls1046ardb/ls1046a/smp
```
```suggestion
identifier: ls1046ardb/ls1046a/smp
```

JiafeiPan replied:
fixed, thanks.

nordicjm commented on boards/nxp/ls1046ardb/ls1046ardb_ls1046a_smp_4cores.yaml:
```diff
@@ -1,4 +1,4 @@
-identifier: nxp_ls1046ardb_smp_4cores
+identifier: nxp_ls1046ardb/ls1046a/smp/4cores
```
```suggestion
identifier: ls1046ardb/ls1046a/smp/4cores
```

JiafeiPan replied:
fixed, thanks.

nordicjm reviewed (commented):
Final change which should fix CI
